src/nemos/base_class.py

```diff
@@ -384,8 +384,23 @@ def _preprocess_fit(
         self._check_input_dimensionality(X, y)
         self._check_input_n_timepoints(X, y)
 
-        check_invalid_entry(X, "X")
-        check_invalid_entry(y, "y")
+        valid_x, err_x = check_invalid_entry(X, "X")
+        valid_y, err_y = check_invalid_entry(y, "y")
+
+        # warn the user of the dropped samples.
+        if err_x:
+            message = err_x.join(" Dropping corresponding samples.")
+            warnings.warn(message=message, category=UserWarning)
+        if err_y:
+            message = err_y.join(" Dropping corresponding samples.")
+            warnings.warn(message=message, category=UserWarning)
+
```

Comment:
This structure feels weird: why have the check and drop happen in the utility function but the warning here? Why not do them all in the utility function? (would clean this code up and avoid the potential problem of developers forgetting to do anything with the message from `check_invalid_entry` ).

Reply:
Also, message should make clear that we're dropping the same samples from *both* arrays

Reply:
(that message might need to be in this function)

Reply:
just realized that the dropping does happen here -- see below for proposal on some restructure of this